lib/config.js

```diff
@@ -0,0 +1,31 @@
+'use strict'
+
+const DOT_REGEXP = /\./g
+
+module.exports = {
+  getConfig
+}
+
+function getConfig(event) {
+  const pkg = require('../package.json')
+  const config = {
+    UserAgent: `${pkg.name}/${pkg.version}`
+  }
+
+  if (process.env.LOGDNA_KEY) config.key = process.env.LOGDNA_KEY
+  if (process.env.LOGDNA_HOSTNAME) config.hostname = process.env.LOGDNA_HOSTNAME
+  if (process.env.LOGDNA_TAGS) {
+    config.tags = process.env.LOGDNA_TAGS.split(',').map((tag) => {
+      return tag.trim()
+    }).join(',')
+  }
+
+  if (event && event.Records && Array.isArray(event.Records)
+    && event.Records[0] && event.Records[0].s3 && event.Records[0].s3.bucket
+    && event.Records[0].s3.bucket.name && !config.hostname
+  ) {
+    config.hostname = event.Records[0].s3.bucket.name.replace(DOT_REGEXP, '_')
+  }
```

Comment:
I know you plan on changing this with `env-config`, but probably still relevant:
```js
const bucketName = getProperty(event, 'Records.0.s3.bucket.name')
if (bucketName && !config.hostname) {
  config.hostname = bucketName.replace(DOT_REGEXP, '_')
}
```